src/UI/Root/Components/Header/EnvSelector/EnvironmentSelector.test.tsx

```diff
@@ -131,3 +135,62 @@ test("GIVEN EnvironmentSelector and environments with identical names WHEN user
 
   expect(selectedEnv).toEqual(envB.id);
 });
+
+const setup = (config: KeycloakAuthConfig | LocalConfig | undefined) => {
+  const queryClient = new QueryClient();
+  return (
+    <MemoryRouter>
+      <QueryClientProvider client={queryClient}>
+        <AuthProvider config={config}>
+          <AuthTestWrapper dependencies={dependencies}>
+            <EnvironmentSelector
+              environments={RemoteData.success([])}
+              onSelectEnvironment={() => {
+                return;
+              }}
+              selectedEnvironment={undefined}
+            />
+          </AuthTestWrapper>
+        </AuthProvider>
+      </QueryClientProvider>
+    </MemoryRouter>
+  );
+};
+
+test("GIVEN EnvironmentSelector WHEN jwt auth is enabled will display fetched username on load", async () => {
+  const server = setupServer(
+    http.get("/api/v2/current_user", async () => {
+      return HttpResponse.json({
+        data: {
+          username: "test_user",
+        },
+      });
+    }),
+  );
+  server.listen();
+  render(setup({ method: "jwt" }));
+
+  await waitFor(() => {
+    expect(screen.getByText("test_user")).toBeVisible();
+  });
+  server.close();
+});
+
+test("GIVEN EnvironmentSelector WHEN jwt auth is enabled will won't fetched username on load", async () => {
```

Comment:
The test case has not the jwt auth enabled, it has the database auth enabled.

The text says you'd check whether the user is fetched or not, but you can't exactly test that because you can add as many handlers to your `setupServer()` method, if the call is not triggered it simply won't fire the handler. You can see them as eventlisteners, but for api-calls.

Also, considering you're setting up a database auth, this means the you'd expect the text in the env selector to be something like no user, or null, or whichever the fallback is in that scenario. 

This test is for me a false positive and doesn't add anything to the coverage.

You'd better test a failing path, where the api endpoint returns no user. A case that should result in a failure, since this is not supposed to happen.

Reply:
added failing path